Sources/CryptoSwift/BlockMode/CFB.swift

```diff
@@ -43,28 +52,55 @@ struct CFBModeWorker: BlockModeWorker {
   let blockSize: Int
   let additionalBufferSize: Int = 0
   private let iv: ArraySlice<UInt8>
+  private let segmentSize: CFB.SegmentSize
   private var prev: ArraySlice<UInt8>?
 
-  init(blockSize: Int, iv: ArraySlice<UInt8>, cipherOperation: @escaping CipherOperationOnBlock) {
+  init(blockSize: Int, iv: ArraySlice<UInt8>, segmentSize: CFB.SegmentSize, cipherOperation: @escaping CipherOperationOnBlock) {
     self.blockSize = blockSize
     self.iv = iv
+    self.segmentSize = segmentSize
     self.cipherOperation = cipherOperation
   }
 
   mutating func encrypt(block plaintext: ArraySlice<UInt8>) -> Array<UInt8> {
-    guard let ciphertext = cipherOperation(prev ?? iv) else {
-      return Array(plaintext)
+    // CFB128
+    if segmentSize == .cfb128 {
+      guard let ciphertext = cipherOperation(prev ?? iv) else {
+        return Array(plaintext)
+      }
+      self.prev = xor(plaintext, ciphertext.slice)
+      return Array(self.prev ?? [])
+    }
+    // CFB8
+    else if segmentSize == .cfb8 {
+      guard let ciphertext = cipherOperation(prev ?? iv) else {
+        return Array(plaintext)
+      }
+      let result = [Array(plaintext)[0] ^ Array(ciphertext)[0]]
+      self.prev = Array((prev ?? iv).dropFirst()) + [result[0]]
```

Comment:
isn't it... slow? Did you check maybe how it compares to CFB128?

Reply:
I understand it is slower than CFB128, but CFB8 is designed to encrypt byte per byte, and so the IV is shifted with each new encrypted byte each time. That's also why the block size is 1 byte and not 16 bytes.